monai/metrics/wrapper.py

```diff
@@ -0,0 +1,290 @@
+        y = torch.tensor([[[[1, 0], [0, 1]], [[0, 1], [0, 0]], [[0, 0], [1, 0]]]])
+        print(metric(y_pred, y))
+
+        # aggregate
+        # shape ([batch=2, channel=1])
+        print(metric.aggregate(reduction="none"))  # tensor([[0.2727], [0.6000]])
+
+        # reset
+        metric.reset()
+
+    """
+
+    def __init__(
+        self,
+        metric_name: str,
+        include_background: bool = True,
+        reduction: Union[MetricReduction, str] = MetricReduction.MEAN,
+        get_not_nans: bool = False,
+        smooth_dr: float = 1e-5,
+    ) -> None:
+        super().__init__(
+            metric_name=metric_name,
+            include_background=include_background,
+            reduction=reduction,
+            get_not_nans=get_not_nans,
+        )
+        self.smooth_dr = smooth_dr
+
+    def _compute_tensor(self, y_pred, y):
+        """Computes a categorical (multi-class) MetricsReloaded metric from a batch of
+        predictions and references.
+
+        Args:
+            y_pred: Prediction with dimensions (batch, channel, *spatial). The values should be
+                one-hot encoded and binarized.
+            y: Ground-truth with dimensions (batch, channel, *spatial). The values should be 1
+                one-hot encoded and binarized.
+
+        Raises:
+            ValueError: when `y` or `y_pred` is not a binarized tensor.
+            ValueError: when `y_pred` has less than three dimensions.
+
+        """
+        # Preprocess
+        y_pred, y, device = self.prepare_onehot(y_pred, y)
+
+        # Sanity check
+        dims = y_pred.ndimension()
+        if dims < 3:
+            raise ValueError(f"y_pred should have at least 3 dimensions (batch, channel, spatial), got {dims}.")
+
+        num_classes = y_pred.shape[1]
+
+        # Reshape and permute for compatible dimension with MetricsReloaded
+        y_pred = y_pred.reshape(y_pred.shape[0], y_pred.shape[1], -1)
+        y_pred = y_pred.permute((0, 2, 1))
+        y = y.reshape(y.shape[0], y.shape[1], -1)
+        y = y.permute((0, 2, 1))
+        dims = y_pred.ndimension()
+
+        # To numpy array
+        y_pred = convert_to_numpy(y_pred)
+        y = convert_to_numpy(y)
+
+        # Create categorical pairwise metric object
+        bpm = MultiClassPairwiseMeasures(
+            y_pred,
+            y,
+            axis=tuple(range(1, dims)),
+            smooth_dr=self.smooth_dr,
+            list_values=list(range(num_classes)),
+            is_onehot=True,
+        )
+
+        # Is requested metric available?
+        if self.metric_name not in bpm.metrics:
+            raise ValueError(f"Unsupported metric: {self.metric_name}")
+
+        # Compute metric
+        metric = bpm.metrics[self.metric_name]()
+
+        # Put back singleton channel dimension
+        metric = metric[..., None]
+
+        # Return metric as numpy array
+        return convert_to_tensor(metric, device=device)
```

Comment:
The comment is "as numpy array" but the function is "to_tensor".

Thanks.

Reply:
Fixed, thanks!